### PR TITLE
Set up associations

### PR DESCRIPTION
**Pull request by JacintaNgwu**
- Created a class Classroom with the @label instance variable, initialized in the constructor, and setter and getter for @label in the attr_accessor.
- Created the has-many/belongs-to relationship between Classroom and Student. 
- Created a class Book with the  @title and @author instance variables, initialized in the constructor, and setters and getters for instance variables in the attr_accessor.
- Created a class Rental with the  @date instance variable, initialized in the constructor, and setter and getter for @date in the attr_accessor.
- Created the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental.

### CODE REVIEW

File: rental_class.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+class Rental
+  attr_accessor :date, :person, :book
+
+  def initialize(date, person, book)
+    @date = date
+    @person = person
+    @book = book
+    @person.rentals = []
```
Comment (Ghiftee) [inline rental_class.rb]: - [ ] Like book, kindly ensure that the rental is adding itself to the @person achieving the 'belongs-to' association. e.g
`person.rentals << self`

File: person_class.rb [hunk 2]
```diff
@@ -13,6 +13,14 @@ def can_use_services?
     is_of_age? || @parent_permission
   end
 
+  def correct_name
+    @name
+  end
+
+  def add_rental(date, person)
+    Rental.new(date, self, person)
+  end
```
Comment (Ghiftee) [inline person_class.rb]: - [ ] For rentals to be created, ```date``` and ```book``` needs to be passed in as parameters, kindly change ```def add_rental(date, person)``` to ```def add_rental(date, book)```
- [ ] Adding ```self``` in ```Rental.new(date, self, person)``` already adds ```person```  to it so kindly replace ```person``` with ```book```

File: person_class.rb [hunk 3]
```diff
@@ -1,6 +1,6 @@
 class Person
   attr_reader :id
-  attr_accessor :name, :age
+  attr_accessor :name, :age, :rentals
```
Comment (Ghiftee) [inline person_class.rb:3]: - [ ] After creating getters and setters method for rentals, it then needs to be initialized. You can read more about it [here](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/class_and_object_for_irb.md#constructor-and-instance-vars)

**Review by Ghiftee — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️♻️
Hello @JacintaNgwu 👋🏽 

Good job working on the project so far! 
To highlight 👍🏽 
✔️ README is professional
✔️ No linter errors
✔️ Associations are set up

However, there are some issues that you need to work on to go to the next project but you are almost there!



## Required Changes ♻️
- [ ] When setting the classroom for a student, kindly ensure that it also adds it to the classrooms' students e.g 
```
class Student < Person
  attr_reader :classroom
  .
  .
  def classroom=(classroom)
      @classroom = classroom
      classroom.students.push(self) unless classroom.students.include?(self)
  end
end
```

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by arslanbisharat — APPROVED** [review]
Hi @JacintaNgwu,
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- The Pull Request has a proper title and description. ✔️ 
- You have added a descriptive Readme file. Good Job. ✔️ 
- There is no linter error is present. ✔️ 
- Project has met all the technical aspects you needed for this project. Keep rocking! ✔️ 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**